app/views/application/_footer.html.slim

```diff
@@ -10,9 +10,9 @@
         = ' '
         = link_to t('.hasso_plattner_institute'), 'https://hpi.de/'
         = ' - '
-        = link_to t('.imprint'), 'https://open.hpi.de/pages/imprint'
+        = link_to t('.imprint'), t('.imprint_link')
```

Comment:
We (or rather I) decided with this current implementation to be in line with other, similar parts of the app. For example, we also have some other localizable elements, where we are not directly _showing_ the text to users:

https://github.com/openHPI/codeharbor/blob/a805c1a9f74f432dee9685903d99e9450bcda9e7/config/locales/en/views/home.yml#L86-L88

Similarly, we are now localizing the `imprint_link` in the different languages, which wouldn't be possible with a static config:

```lml
imprint_link: https://open.hpi.de/pages/imprint?locale=de
```

Reply:
We could also store the link in the settings and then add the locale parameter in the view.

Reply:
That's correct, indeed. The only assumption would be that we use the same locale parameters between both apps (which _should_ be the case, but I can tell that there might be some instances where this assumption is not correct. For example, consider the following code:

```rb
# The incorrect Chinese language code `cn` is still used in the platform.
# To be able to use the correct code `zh` in the data, it is mapped here.
h['cn'] = h['zh'] if h.key?('zh')
```

As you can see, there is some issue with the Chinese locale, incorrectly referred to with the locale code `cn` (inferred from the domain). However, the correct one would be `zh`. Unfortunately, however, the correct locale is not consistently used, and you can only use the incorrect https://open.hpi.de/?locale=cn rather than the correct https://open.hpi.de/?locale=zh for Chinese. Given this real-world example, automatically constructing such a locale string is difficult, and there are more examples [e.g., Portuguese vs. Brazilian-Portuguese, ...]. Sure, it will work as long as some assumptions are met, but the current one only relies on one: A human entering the correct link 😉.)

Having that said, I would probably just stick with the current implementation.